CMakePresets.json

```diff
@@ -0,0 +1,62 @@
+{
+  "version": 3,
+  "cmakeMinimumRequired": {
+    "major": 3,
+    "minor": 21,
+    "patch": 0
+  },
```

Comment:
It also bugs me that this file needs to live at the root, instead of `build_tools/cmake`. At least it sorts lexicographically next to `CMakeLists.txt`

https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html

Reply:
I'm certainly bugged by worse things about this mechanism...